flocker/node/agents/ebs.py

```diff
@@ -997,23 +1019,16 @@ def _attach_ebs_volume(self, volume_id, instance_id, device):
         return volume.attach_to_instance(
             InstanceId=instance_id, Device=device)
 
-    def _next_device(self, instance_id, volumes, devices_in_use):
+    def _next_device(self, devices_in_use):
         """
-        Get the next available EBS device name for a given EC2 instance.
-
-        Algorithm:
-        1. Get all ``Block devices`` currently in use by given instance:
-            a) List all volumes visible to this instance.
-            b) Gather device IDs of all devices attached to (a).
-        2. Devices available for EBS volume usage are ``/dev/sd[f-p]``.
-           Find the first device from this set that is currently not
-           in use.
+        Get the next available EBS device name for this EC2 instance.
+
+        Devices available for EBS volume usage are ``/dev/sd[f-p]``.
+        Find the first device from this set that is currently not
+        in use.
         XXX: Handle lack of free devices in ``/dev/sd[f-p]`` range
         (see https://clusterhq.atlassian.net/browse/FLOC-1887).
 
-        :param unicode instance_id: EC2 instance ID.
-        :param volumes: Collection of currently known
-            ``BlockDeviceVolume`` instances.
         :param set devices_in_use: Unicode names of devices that are
```

Comment:
The passed in value is a set of devices in the form `"/dev/sda"`. Which doesn't match the internally used format of `"sdb"`.

1. This suggests insufficient test coverage.
2. However... the reason we do the ignore thing is in case we attach and it fails. Since we're looking at all local devices in this code path it seems like we don't need to know that an attach failed because of an attached volume, we'll see it next time we do `lsblk`. So maybe this parameter can just be dropped?